helix-rest/src/main/java/org/apache/helix/rest/server/resources/helix/PropertyStoreAccessor.java

```diff
@@ -0,0 +1,68 @@
+package org.apache.helix.rest.server.resources.helix;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLDecoder;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Response;
+
+import org.apache.helix.AccessOption;
+import org.apache.helix.HelixException;
+import org.apache.helix.PropertyPathBuilder;
+import org.apache.helix.ZNRecord;
+import org.apache.helix.manager.zk.ZkBaseDataAccessor;
+import org.apache.helix.store.HelixPropertyStore;
+import org.apache.helix.store.zk.ZkHelixPropertyStore;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Path("/clusters/{clusterId}/propertyStore")
+public class PropertyStoreAccessor extends AbstractHelixResource {
+  private static Logger LOG = LoggerFactory.getLogger(PropertyStoreAccessor.class);
+
+  @GET
+  public Response getPropertyByPath(@PathParam("clusterId") String clusterId,
+      @QueryParam("path") String urlEncodedPath) {
```

Comment:
@i3wangyi I was working on this a few weeks back. My idea is to use regex to parse the propertyStore path using `{path: .+}` so `/propertyStore/aa/bb/cc` would be parsed to get path = "/aa/bb/cc". Path doesn't have to be encoded or decoded.

Reply:
The problem is transmitting the "/aa/bb/cc" in the Http request URL. Something like this `"http://endpoint.linkedin.biz/clusters/CLUSTER/propertyStore/a/b/c/d"` doesn't look a legit case to me. Do you think the regex can help stop the directory like "/"?

Reply:
I think it looks fine. For example, in Hadoop WebHDFS API, the path slash is also presented in the api url. 
```
curl -i -b cookie.jar "http://host:port/webhdfs/v1/tmp/file.txt?op=GET_BLOCK_LOCATIONS"
```
About encoding the path, if only the web ui uses the api, it is ok, as the front end could encode the path. But I don't think it is convenient for users to use the api manually, if users still have to run a script like python to encode the path. I always believe we need to consider user friendly.

Reply:
Thanks for the example of Hadoop API. I will make the change see if the PathParam is a doable approach (it should be since others already implement it)